tests/integration/framework/client/etcd.go

```diff
@@ -0,0 +1,66 @@
+/*
+Copyright 2024 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+type EtcdClient struct {
+	client *clientv3.Client
+}
+
+// Returns an adapted Etcd client for tests.
+func Etcd(t *testing.T, cfg clientv3.Config) *EtcdClient {
+	etcdClient, err := clientv3.New(cfg)
+	require.NoError(t, err)
+	t.Cleanup(func() { require.NoError(t, etcdClient.Close()) })
+
+	return &EtcdClient{etcdClient}
+}
+
+func (c *EtcdClient) ListAllKeys(ctx context.Context, prefix string) ([]string, error) {
```

Comment:
This is a handy helper method to make sure the list of keys on Etcd consider pagination and iterates on them.